### PR TITLE
Move group with namespace

### PR DESCRIPTION
**Pull request by neka-nat**
### Description
When using MoveGroupCommander in rospy, I made it possible to give namespace arguments to MoveGroupCommander so that we can connect to move_group actions even if namespace is set in move_group.
Related to #757.

### CODE REVIEW

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     attached_object_publisher_ = node_handle_.advertise<moveit_msgs::AttachedCollisionObject>(
         planning_scene_monitor::PlanningSceneMonitor::DEFAULT_ATTACHED_COLLISION_OBJECT_TOPIC, 1, false);
 
-    current_state_monitor_ = getSharedStateMonitor(robot_model_, tf_, node_handle_);
+    current_state_monitor_ = getSharedStateMonitor(opt_.robot_description_, robot_model_, tf_, node_handle_);
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp]: i don't understand why this change is necessary - this slows down the code by requiring a string to be parsed rather than using the pre-constructed RobotModel class

File: moveit_ros/planning_interface/common_planning_interface_objects/include/moveit/common_planning_interface_objects/common_objects.h [hunk 2]
```diff
@@ -56,7 +56,8 @@ robot_model::RobotModelConstPtr getSharedRobotModel(const std::string& robot_des
   @param tf
   @return
  */
-planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot_model::RobotModelConstPtr& kmodel,
+planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const std::string& robot_description,
+                                                                     const robot_model::RobotModelConstPtr& kmodel,
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/common_planning_interface_objects/include/moveit/common_planning_interface_objects/common_objects.h]: you shouldn't need to pass both the robot_description and kmodel... they do the same thing

File: moveit_ros/planning_interface/common_planning_interface_objects/include/moveit/common_planning_interface_objects/common_objects.h [hunk 3]
```diff
@@ -67,7 +68,8 @@ planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot
   @param nh A ros::NodeHandle to pass node specific configurations, such as callbacks queues.
   @return
  */
-planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot_model::RobotModelConstPtr& kmodel,
+planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const std::string& robot_description,
+                                                                     const robot_model::RobotModelConstPtr& kmodel,
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/common_planning_interface_objects/include/moveit/common_planning_interface_objects/common_objects.h]: same

File: moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp [hunk 4]
```diff
@@ -102,25 +102,27 @@ robot_model::RobotModelConstPtr getSharedRobotModel(const std::string& robot_des
   }
 }
 
-planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot_model::RobotModelConstPtr& kmodel,
+planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const std::string& robot_description,
+                                                                     const robot_model::RobotModelConstPtr& kmodel,
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: same

File: moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp [hunk 5]
```diff
@@ -102,25 +102,27 @@ robot_model::RobotModelConstPtr getSharedRobotModel(const std::string& robot_des
...
                                                                      const boost::shared_ptr<tf::Transformer>& tf)
 {
-  return getSharedStateMonitor(kmodel, tf, ros::NodeHandle());
+  return getSharedStateMonitor(robot_description, kmodel, tf, ros::NodeHandle());
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: same

File: moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp [hunk 6]
```diff
@@ -102,25 +102,27 @@ robot_model::RobotModelConstPtr getSharedRobotModel(const std::string& robot_des
...
 }
 
-planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const robot_model::RobotModelConstPtr& kmodel,
+planning_scene_monitor::CurrentStateMonitorPtr getSharedStateMonitor(const std::string& robot_description,
+                                                                     const robot_model::RobotModelConstPtr& kmodel,
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: same

File: moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp [hunk 7]
```diff
@@ -102,25 +102,27 @@ robot_model::RobotModelConstPtr getSharedRobotModel(const std::string& robot_des
...
                                                                      const boost::shared_ptr<tf::Transformer>& tf,
                                                                      ros::NodeHandle nh)
 {
   SharedStorage& s = getSharedStorage();
   boost::mutex::scoped_lock slock(s.lock_);
-  if (s.state_monitors_.find(kmodel->getName()) != s.state_monitors_.end())
-    return s.state_monitors_[kmodel->getName()];
+  if (s.state_monitors_.find(robot_description) != s.state_monitors_.end())
+    return s.state_monitors_[robot_description];
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: this is a big change in behavior.. getName() usually returns e.g. ``pr2`` whereas robot_description ``robot_description``

Comment (neka-nat) [reply moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: I made such a change because of passing the included test code.
This test code makes it possible to control two robots via moveit.
In the test code, it sets two robot models to rosparam.
The first one is "/robot_description", the second one is "/test/robot_description".

In the original code, `kmodel->getName()` of the both robots becomes "fanuc", and it returns the same state to the two robots.
In the changed code, I use `robot_description` so that keys are unique.

Comment (neka-nat) [reply moveit_ros/planning_interface/common_planning_interface_objects/src/common_objects.cpp]: I am also aware that it is a big change, but I definitely want to control multiple robots with moveit.
I would like to try it if you have an alternative idea.

**Comment by willcbaker** [comment]
#835

**Comment by neka-nat** [comment]
Thanks, @willcbaker !
